### PR TITLE
fix: getComposer method

### LINKED ISSUE #76: getComposer is broken in 2.x-dev
#72 is using the `/p2/<package>~dev.json` file for the `getComposer` method and adds a `getComposerLite` method for `/p2/<package>.json`. But this is **not** equivalent to the behavior of the 1.x version, due to a misunderstanding of the way those files look like.
The `~dev.json` file contains *only* versions for branches while the `.json` file contains the tags (this is done to improve caching, as the tag file is invalidated only when a release is done while the branch file is invalidated on each push, and composer does not need to dev versions by default).

To reproduce the 1.x behavior, the SDK would need to load both files and then merge the version lists for each package name.
And `lite` is a bad description. `getComposerReleases` would be a better name.

### PR DESCRIPTION
**Pull request by JellyBellyDev**
Fixed #76 
I hope :)

### CODE REVIEW

**Comment by JellyBellyDev** [comment]
I don't know why the CI doesn't pass, suggestions?

File: src/Packagist/Api/Client.php [hunk 1]
```diff
@@ -192,6 +206,34 @@ protected function respond(string $url)
...
+
+        foreach ($response1['packages'] as $k1 => $package1) {
+            foreach ($response2['packages'] as $k2 => $package2) {
+                if ($k1 === $k2) {
+                    foreach ($package2 as $version) {
+                        $response1['packages'][$k1][] = $version;
+                    }
+                }
+            }
+        }
```
Comment (robbieaverill) [inline src/Packagist/Api/Client.php:232]: I think there is probably a performance optimisation to be made here, to reduce the level of loop nesting

**Review by robbieaverill — COMMENTED** [review]
Hey @JellyBellyDev, I haven't forgotten about this PR, and will get to making some suggestions shortly. Thanks for making it, and sorry for the delay!

File: src/Packagist/Api/Client.php (hunk 1 above)
Comment (robbieaverill) [reply src/Packagist/Api/Client.php:232]: I'll take a look at this today and see if I can make a suggestion. Thanks again for the PR, and sorry again for taking so long to get back to you.

Comment (robbieaverill) [reply src/Packagist/Api/Client.php:232]: @JellyBellyDev from what I can tell the `$package2` we receive in this loop has a different data structure depending on the version, i.e. the first time it contains `name`, `description`, etc, then the following times it doesn't - only containing `version` etc. Did you come across that in your testing?

Comment (JellyBellyDev) [reply src/Packagist/Api/Client.php:232]: Yep, the data structure it's very strange and I have tried to manage all the cases...

**Comment by robbieaverill** [comment]
Hi @JellyBellyDev, still working on this - I've made a couple of potential changes and am working on the tests, will post an update when I can.

**Comment by robbieaverill** [comment]
I've tested this against the behaviour in 1.7.x using the original `getComposer()` method added in https://github.com/KnpLabs/packagist-api/commit/37a9b1ec6f20e6e7c3ad8648f263f1dfd2e1662a, and have put the original doc block (slightly adjusted) from that commit back in place. All looks good to me now. Thanks for your contribution @JellyBellyDev and sorry again for the time it took for me to review it.

**Review by robbieaverill — APPROVED** [review]

File: src/Packagist/Api/Client.php [hunk 2]
```diff
@@ -216,19 +224,19 @@ protected function respond(string $url)
...
+        $response2 = $this->parse((string) $response2);
+
+        foreach ($response1['packages'] as $name => $package1) {
+            if (empty($response2['packages'][$name])) {
+                continue;
             }
+            $response1['packages'][$name] = [
+                ...$response1['packages'][$name],
+                ...$response2['packages'][$name],
+            ];
```
Comment (JellyBellyDev) [inline src/Packagist/Api/Client.php:239]: Very nice!!! GJ!!! 🚀

**Pull request merged**